src/toil/resource.py

```diff
@@ -24,19 +24,33 @@
 from io import BytesIO
 from pydoc import locate
 from tempfile import mkdtemp
-from typing import Any
 from urllib.error import HTTPError
 from urllib.request import urlopen
 from zipfile import ZipFile
 
+from typing import (TYPE_CHECKING,
+                    Optional,
+                    Any,
+                    Callable,
+                    Union,
+                    IO,
+                    Type,
+                    Tuple,
+                    Sequence,
+                    TextIO)
+
 from toil import inVirtualEnv
 from toil.lib.iterables import concat
 from toil.lib.memoize import strict_bool
 from toil.lib.retry import ErrorCondition, retry
 from toil.version import exactPython
 
-logger = logging.getLogger(__name__)
+if TYPE_CHECKING:
+    from toil.jobStores.abstractJobStore import AbstractJobStore
+    from types import ModuleType
+    from io import FileIO
```

Comment:
We might need to import these unconditionally; I think we only want to use `TYPE_CHECKING` to break circular imports, and then the things need to be quoted strings when actually used in types, since they won't be defined at runtime.